### PR TITLE
Feature/return good values

### PR DESCRIPTION
**Pull request by Pfontaine**
This is what this microservice must looks like.
Test for flask route is missing, @clairedupuich you can try to add it.

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        skip = page_size * (page_on - 1)
+        coll = self.__get_collection()
+
+        dict_filtre = self.__make_filter(args)
+
+        page_record = coll.find(dict_filtre).limit(page_size).skip(skip)
+        total_record = coll.count_documents(dict_filtre)
+        data_to_respond = ResponseCreator(page_record, total_record, page_on, page_size)
+
+        return flask.Response(json.dumps(data_to_respond.create_response()), mimetype='application/json')
```
Comment (Pfontaine) [inline app/app.py:41]: It's your old code

File: app/app.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        :param args: Flask request args
+        :return: Filters for MongoDB
+        """
+        dict_filtre = {}
+        if args.get("station_id") is not None:
+            dict_filtre['station_uuid'] = bson.Binary.from_uuid(UUID(args.get('station_id')))
+        if args.get("owner_id") is not None:
+            dict_filtre['owner.uuid'] = bson.Binary.from_uuid(UUID(args.get('owner_id')))
+        return dict_filtre
```
Comment (Pfontaine) [inline app/app.py:60]: I've extract the filters to made more easy to change or add filters in the future

File: README.md [hunk 3]
```diff
@@ -1,21 +1,28 @@
-# MayaProtect - Microservice hive_list
```
Comment (Pfontaine) [inline README.md:1]: Use the name in API Definition

File: app/app.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        self.params = params
+        self.app = flask.Flask(__name__)
+        CORS(self.app)
+        self.my_client = pymongo.MongoClient("mongodb://" + params['mongo_host'] + ":" + params['mongo_port'] + "/")
+        self.db = self.my_client[params['mongo_db']]
+        self.default_limit = params['default_limit']
+        # Register routes
+        self.app.add_url_rule('/hives', 'get_hives', self.get_hives, methods=['GET'])
+        # Set CORS
+        self.app.config['CORS_HEADERS'] = 'Content-Type'
```
Comment (Pfontaine) [inline app/app.py:21]: Run and configure flask in the init of the app class

File: app/event.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            raise ValueError("The event must have a created_at")
+
+        if type(event['createdAt']) is not str:
+            raise TypeError("The event created_at must be a string")
+
+        if event['creator'] is None:
+            raise ValueError("The event must have a creator")
+
+        if event['description'] is None:
+            raise ValueError("The event must have a description")
```
Comment (Pfontaine) [inline app/event.py:30]: Don't forget to check if data is present

File: run_test.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    # Create a test suite
+    suite = unittest.TestSuite()
+    # Add all tests to the test suite
+    loader = unittest.TestLoader()
+    suite.addTest(loader.loadTestsFromTestCase(TestEvent))
+    suite.addTest(loader.loadTestsFromTestCase(TestOwner))
+    suite.addTest(loader.loadTestsFromTestCase(TestHive))
+    suite.addTest(loader.loadTestsFromTestCase(TestResponseCreator))
+
+    # suite.addTest(loader.loadTestsFromTestCase(TestApp))
```
Comment (Pfontaine) [inline run_test.py:15]: Don't forget to uncomment this line when you create test for flask route

File: tests/__init__.py [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# from tests.test_app import TestApp
```
Comment (Pfontaine) [inline tests/__init__.py:1]: Don't forget to uncomment this line

**Review by Pfontaine — COMMENTED** [review]
Comments in review

**Review by clairedupuich — APPROVED** [review]

**Pull request merged**